### PR TITLE
feat: add workspace:delete command

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
Allows to delete user's workspace and namespace.
Use should have cluster admin rights to delete namespaces

```bash
$ ./run workspace:list -n abazkoche --access-token=$CHE_ACCESS_TOKEN
  ✔ Verify Kubernetes API...OK (it's OpenShift)
  ✔ Verify if namespace 'abazkoche' exists
  ✔ Retrieving Eclipse Che server URL... http://che-abazkoche.apps.ocp43.codereadyqe.com
  ✔ Eclipse Che status check
  ✔ Get workspaces... done
Id                        Name                          Namespace Status  Created                  Updated                  
workspace0kxaqsiwpe2h77g2 apache-camel-springboot-m83c3 admin-che STOPPED 2020-03-18T14:56:51.675Z                          
workspace8d7pzoq23h9iffiw golang-g2rp3                  admin-che STOPPED 2020-03-18T14:57:03.415Z 2020-03-18T14:57:05.404Z 
 $ ./run workspace:delete -n abazkoche --access-token=$CHE_ACCESS_TOKEN -w workspace0kxaqsiwpe2h77g2
  ✔ Verify Kubernetes API...OK (it's OpenShift)
  ✔ Verify if namespace 'abazkoche' exists
  ✔ Retrieving Eclipse Che server URL... http://che-abazkoche.apps.ocp43.codereadyqe.com
  ✔ Eclipse Che status check
  ✔ Get workspace with id 'workspace0kxaqsiwpe2h77g2'... done
  ✔ Delete workspace with id 'workspace0kxaqsiwpe2h77g2'... done
  ✔ Verify if namespace exists 'admin-che'... found
  ✔ Delete namespace 'admin-che'... done
$ ./run workspace:delete -n abazkoche --access-token=$CHE_ACCESS_TOKEN -w workspace8d7pzoq23h9iffiw
  ✔ Verify Kubernetes API...OK (it's OpenShift)
  ✔ Verify if namespace 'abazkoche' exists
  ✔ Retrieving Eclipse Che server URL... http://che-abazkoche.apps.ocp43.codereadyqe.com
  ✔ Eclipse Che status check
  ✔ Get workspace with id 'workspace8d7pzoq23h9iffiw'... done
  ✔ Delete workspace with id 'workspace8d7pzoq23h9iffiw'... done
  ✔ Verify if namespace exists 'admin-che'... not found
  ↓ Delete namespace [skipped]
```

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/16236

### CODE REVIEW

File: src/commands/workspace/delete.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        ctx.infrastructureNamespaceExists = await kubeHelper.namespaceExist(ctx.infrastructureNamespace)
+        if (ctx.infrastructureNamespaceExists) {
+          task.title = `${task.title}... found`
+        } else {
+          task.title = `${task.title}... not found`
+        }
+      }
+    })
+    tasks.add({
+      title: 'Delete namespace',
```
Comment (AndrienkoAleksandr) [inline src/commands/workspace/delete.ts]: Hello. @tolusha workspace can be configured to start workspace with the same namespace with Che...

Comment (tolusha) [reply src/commands/workspace/delete.ts]: yes. that's why I check
```
skip: ctx => ctx.infrastructureNamespace === flags.chenamespace,
```
if it is the same namespace then I won't delete it

**Comment by sleshchenko** [comment]
@tolusha Are we able to avoid having parameter to the main argument - identifier?
Like kubectl/oc does with everything: `oc get pod my-pod-name`...
I wonder if we could do the same: `chectl workspace:delete workspace123923 -n myche` or even `chectl workspace:delete username/java-mysql -n myche` :rocket:

**Comment by sleshchenko** [comment]
```
>  $ ./run workspace:delete -n abazkoche --access-token=$CHE_ACCESS_TOKEN -w workspace0kxaqsiwpe2h77g2
  ✔ Verify Kubernetes API...OK (it's OpenShift)
  ✔ Verify if namespace 'abazkoche' exists
  ✔ Retrieving Eclipse Che server URL... http://che-abazkoche.apps.ocp43.codereadyqe.com
  ✔ Eclipse Che status check
  ✔ Get workspace with id 'workspace0kxaqsiwpe2h77g2'... done
  ✔ Delete workspace with id 'workspace0kxaqsiwpe2h77g2'... done
  ✔ Verify if namespace exists 'admin-che'... found
  ✔ Delete namespace 'admin-che'... done
```
Is `admin-che` namespace read from workspace configuration?
:warning: I'm not sure if we need even try to delete namespace, maybe only when user specify additional parameter... Like `admin-che` namespace is reused for every `admin`'s namespace. So, it mean that every namespace lose their PVC

**Comment by tolusha** [comment]
@sleshchenko 

> Are we able to avoid having parameter to the main argument - identifier?

That's not possible due to underlying cli tool

**Comment by sleshchenko** [comment]
>That's not possible due to underlying cli tool

@tolusha Are you sure? Seems we just haven't used it before but it's possible. It's called argument while we use only flags https://oclif.io/docs/args

**Comment by tolusha** [comment]
That's much better
```bash
$ ./run workspace:list
Id                        Name      Namespace Status  Created                  Updated                  
workspaceisyay7xhi8euo94t wksp-r1ua che-che   STOPPED 2020-03-19T08:55:23.414Z 2020-03-19T08:55:24.392Z 
[10:58:22] tolusha@fractal ~/projects/chectl/bin (ab/delete) $ ./run workspace:delete -w workspaceisyay7xhi8euo94t --delete-namespace
Workspace with id 'workspaceisyay7xhi8euo94t' deleted.
Namespace 'che-che' deleted.
```

**Comment by tolusha** [comment]
@sleshchenko 

> It's called argument

agree, it is possible. Not sure if it is good practice to use arguments for one command and flags for another. We should be consistent.

**Comment by tolusha** [comment]
[test]

**Comment by tolusha** [comment]
@sleshchenko 
agreed

**Pull request merged**